pkg/connector/role_cache.go

```diff
@@ -61,15 +81,23 @@ func (p *roleCache) buildCache(ctx context.Context) error {
 }
 
 func (p *roleCache) getRoleByFolder(folderId int) []*client.Role {
+	if !p.initialized {
+		return nil
+	}
```

Comment:
I think we should use a sync.RwMutex for every read call, instead of the sync.Mutex that we have

p.mu.RLock()
defer p.mu.RUnlock()